src/main/java/com/flippingutilities/model/HistoryManager.java

```diff
@@ -360,52 +342,67 @@ public void validateGeProperties()
-		removeInvalidatedOfferEvents();
+	public boolean hasOfferInInterval(Instant earliestTime) {
+		if (!hasValidOffers()) {
+			return false;
+		}
+		OfferEvent lastOffer = compressedOfferEvents.get(compressedOfferEvents.size()-1);
+		return lastOffer.getTime().isAfter(earliestTime);
 	}
 
-	public void removeInvalidatedOfferEvents()
+	/**
+	 * Used to mark offers as invalid. These offers are then pruned later. This is the mechanism
+	 * by which offers are deleted.
+	 * @param offerList the offers being invalidated
+	 *
+	 * @param items    The other items in the account's trade list. These items are needed because
+	 * 	               the offer list that we are invalidating may be used in the combination flips of other
+	 * 	               items, so we have to delete all combination flips that use these offers, both in this item
+	 * 	               and in all other items.
+	 */
+	public void deleteOffers(List<OfferEvent> offerList, List<FlippingItem> items)
 	{
-		if (nextGeLimitRefresh == null)
-		{
-			compressedOfferEvents.removeIf(offer -> !offer.isValidOfferEvent());
+		if (offerList.isEmpty()) {
 			return;
 		}
 
-		Instant startOfRefresh = nextGeLimitRefresh.minus(4, ChronoUnit.HOURS);
+		Set<String> idsOfOffersToBeDeleted = offerList.stream().map(OfferEvent::getUuid).collect(Collectors.toSet());
+		List<CombinationFlip> combinationFlipsToBeDeleted = getInvalidatedCombinationFlips(idsOfOffersToBeDeleted);
+		deleteCombinationFlips(combinationFlipsToBeDeleted, items);
 
-		compressedOfferEvents.removeIf(offer -> !offer.isValidOfferEvent() &&
-			(offer.getTime().isAfter(nextGeLimitRefresh) || offer.getTime().isBefore(startOfRefresh)));
+		//TODO just delete the events, no need to mark them as invalid. I want to eventually get rid of the
+		//validOfferEvent field..but it's annoying cause it is already persisted.
+		offerList.forEach(offer -> offer.setValidOfferEvent(false));
+		removeInvalidOffers();
 	}
 
 	/**
-	 * Creates Flips from offers. Flips represent a buy trade followed by a sell trade. A trade is a collection
-	 * of offers from the empty offer to the completed offer. A completed offer marks the end of a trade.
-	 *
-	 * @param earliestTime the time after which trades should be looked at
-	 * @return flips
+	 * Deletes combination flips by deleting the combination flip from the personal or parent combination flip
+	 * of every item in the combination flip
 	 */
-	public List<Flip> getFlips(Instant earliestTime)
-	{
-		ArrayList<OfferEvent> intervalHistory = getIntervalsHistory(earliestTime);
-
-		//group offers based on which account those offers belong to (this is really only relevant when getting the flips
-		//of the account wide tradelist as you don't want to match offers from diff accounts.
-		Map<String, List<OfferEvent>> groupedOffers = intervalHistory.stream().collect(Collectors.groupingBy(OfferEvent::getMadeBy));
+	public void deleteCombinationFlips(List<CombinationFlip> combinationFlips, List<FlippingItem> items) {
```

Comment:
The items are needed because when we delete a CombinationFlip, we need to delete that same CombinationFlip from the multiple items it could be stored in. 

For example, you have a FlippingItem that represents a Blowpipe (contains all your trades for blowpipes) and another for Tanzanite Fang (contains all your trades for TFs).

You create a combination flip using an offer event for the Blowpipe and an offer event for the TF. This combination flip now exists in the HistoryManager of the Blowpipe FlippingItem in the `personalCombinationFlips` field. This same combination flip ALSO exists in the `parentCombinationFlips` field in the TF. So when want to delete that CombinationFlip, we need to delete it from BOTH the items, hence the need for passing in the FlippingItems.

The combinations themselves only store offers. They don't store the FlippingItems (which contain HistoryManagers which are actually the thing that contain the CombinationFlips).